api/views/diagnosticimport.py

```diff
@@ -570,6 +581,99 @@ def _column_count_error_message(self, row):
         return f"Données manquantes : au moins 13 colonnes attendues, {len(row)} trouvées. Si vous voulez importer que la cantine, veuillez changer le type d'import et réessayer."
 
 
+class CCImportMixin:
+    total_value_idx = 14
+    year_idx = 13
+    cc_satellite_dict = {}
+
+    def _validate_row(self, row):
+        is_central_kitchen = CCImportMixin._is_central_kitchen(row)
+        last_mandatory_column = self.manager_column_idx - 1
+
+        if is_central_kitchen:
+            if len(row) < self.year_idx + 1:
+                raise IndexError()
+            elif len(row) > self.final_value_idx + 1:
+                raise FileFormatError(
+                    detail=f"Format fichier : {self.final_value_idx + 1} colonnes attendues, {len(row)} trouvées."
+                )
+
+        elif len(row) < last_mandatory_column:
+            raise IndexError()
+
+    def _treat_csv_file(self, file):
+        return_value = super()._treat_csv_file(file)
+        self._clean_removed_satellites()
+        return return_value
+
+    def _clean_removed_satellites(self):
+        for central_cuisine_siret in self.cc_satellite_dict:
+            satellite_sirets = self.cc_satellite_dict[central_cuisine_siret]
+            Canteen.objects.filter(central_producer_siret=central_cuisine_siret).exclude(
+                siret__in=satellite_sirets
+            ).update(central_producer_siret=None)
+
+    def _validate_diagnostic(self, row):
+        is_central_kitchen = CCImportMixin._is_central_kitchen(row)
+        if is_central_kitchen:
+            return super()._validate_diagnostic(row)
+        return None, None, None
+
+    def _update_or_create_canteen(
```

Comment:
est-ce qu'on peut modifier la raison pour le changement dans l'historique ? Comme ça on comprendra facilement si on a des questions sur pourquoi un satellite a changé surtout quand le gestionnaire de la CC ne gère pas le satellite

Reply:
History changes in https://github.com/betagouv/ma-cantine/pull/2750/commits/edabaa090fe98e2b00b4efdada2f59a82d0fcd1a and https://github.com/betagouv/ma-cantine/pull/2750/commits/7eb3f6a83a9377d8cf50c95be9576793b98fa4f3

![image](https://github.com/betagouv/ma-cantine/assets/1225929/b0108b25-0be6-4b89-99ca-685a62bb7c66)
![image](https://github.com/betagouv/ma-cantine/assets/1225929/68e62913-6a5c-4419-b1c1-b4ebb1912207)